### DISCUSSION TITLE
Secret for Signoz

### DISCUSSION THREAD

**Discussion by bdronneau** [SigNoz/signoz#1007] [Q&A] [answered]
Hello,

Thank for the work done here 🚀 
I'm currently testing to put in place signoz on k8s, the documentation and helm chart are quite clear.
But I didn't see in the configuration any "secret/token" in order to protect the endpoint.

I have missing something in the configuration or it's not possible for now ?

Best

**Comment by prashant-shahi** [comment]
Hello @bdronneau!!

We use `ClusterIP` by default so all SigNoz components remain accessible only within the k8s cluster.

> protect the endpoint.

If you meant to protect SigNoz `frontend` and `query-service` public access. The Auth feature is being worked on right now.
You can track it here: https://github.com/SigNoz/signoz/pull/874

**Comment by prashant-shahi** [comment]
@bdronneau If you meant to protect the SigNoz `Otel-Collector`, we have discussed it internally and realised its importance. That task will be picked up in soon future.

**Comment by bdronneau** [comment]
Hi,

Indeed if all stuff are in same cluster, using clusterip is enough. But I
prefer to have separate clusters from tooling / monitoring and running
(apps production).

If you are working on, it's perfect =)

Le jeu. 21 avr. 2022 à 23:53, Prashant Shahi ***@***.***> a
écrit :

**Reply by pranay01** [reply]
hey @bdronneau the login + auth feature is shipped now in this release - https://github.com/SigNoz/signoz/releases/tag/v0.8.0

Do check it out and let us know of any feedback

**Reply by bdronneau** [reply]
Hi,
Wonderful ! I'm currently instrumenting a django app (not fully fonctionnal) so this will be the next step.

**Reply by pranay01** [reply]
Cool, let us know how it goes - and drop by in our slack community if you have question https://signoz.io/slack

**Reply by bdronneau** [reply]
Hi,

Far I understand this [authentification](https://signoz.io/docs/userguide/authentication/) feature it's for the end user, not for the application for send metrics.

**Reply by pranay01** [reply]
yes, that's accurate @bdronneau

**Comment by bdronneau** [comment, marked as answer]
I close this discussion in indeed, it's better to leave collector doing is best like collecting and make basic auth with NGINX if public access

**Reply by drjasonharrison** [reply]
For Enterprise users there will be the desire to have admin vs standard user accounts.
